src/lib.rs

```diff
@@ -272,6 +302,95 @@ pub enum CargoOpt {
     SomeFeatures(Vec<String>),
 }
 
+/// Profile settings used to determine which compiler flags to use for a
+/// target.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ArtifactProfile {
+    /// Optimization level. Possible values are 0-3, s or z.
+    pub opt_level: String,
+    /// The amount of debug info. 0 for none, 1 for limited, 2 for full
+    pub debuginfo: Option<u32>,
+    /// State of the `cfg(debug_assertions)` directive, enabling macros like
+    /// `debug_assert!`
+    pub debug_assertions: bool,
+    /// State of the overflow checks.
+    pub overflow_checks: bool,
+    /// Whether this profile is a test
+    pub test: bool,
+    #[doc(hidden)]
+    #[serde(skip)]
+    __do_not_match_exhaustively: (),
+}
+
+/// A compiler-generated file.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Artifact {
+    /// The workspace member this artifact belongs to
+    pub package_id: WorkspaceMember,
+    /// The target this artifact was compiled for
+    pub target: Target,
+    /// The profile this artifact was compiled with
+    pub profile: ArtifactProfile,
+    /// The enabled features for this artifact
+    pub features: Vec<String>,
+    /// The full paths to the generated artifacts
+    pub filenames: Vec<String>,
```

Comment:
Might those be better as PathBuf than as String?

Reply:
Imho workspace_root should also be changed, but maybe in a different PR. @oli-obk What do you think?

Reply:
@konstin Oliver is away until October.